app/services/flow/base_step.rb

```diff
@@ -26,8 +26,18 @@ def mark_step_incomplete(step = nil)
       flow_session.delete(klass.to_s)
     end
 
+    def self.acceptable_response_object?(obj)
+      obj.is_a?(FormResponse) || obj.is_a?(DocAuth::Response)
+    end
+
     private
 
+    def create_response(form_submit_response, call_response)
+      return form_submit_response unless BaseStep.acceptable_response_object?(call_response)
+      call_response.extra.merge!(form_submit_response.extra)
+      call_response
```

Comment:
yeah I think this could be replaced with `call_response.merge(form_submit_response)` if we added `#merge` to FormResposne --- however one caveat is that `DocAuth::Response` has errors as an array but `FormResponse` has errors as a hash --- so we'll need to make a call about what to do in that case

Reply:
yeah that array/hash has been vexing me